.travis.yml

```diff
@@ -325,6 +372,21 @@ py36_linux_build_engine: &py36_linux_build_engine
     - CACHE_NAME=linuxpexbuild.py36
     - BOOTSTRAPPED_PEX_KEY_SUFFIX=py36.linux
 
+# NB: we do not use Docker for Py37 because Centos6 has too outdated of OpenSSL.
+# See https://github.com/pantsbuild/pants/issues/7421. We will want to change this
+# shard back to Docker once we have a Centos7 base image.
```

Comment:
It may be worth creating this docker image now, and using it, so that the future upgrade is just a matter of switching the default image from 6 to 7 (which we've tested and know works), instead of introducing docker here, but not going to block merging on the idea if you don't want to :)

Reply:
I had originally proposed doing this, but @stuhood and I decided to go with this workaround for now since we’re introducing so many changes to CI, eg default daemon. 

After this gets merged, I agree we should use this shard to iterate on the Centos7 image and get things working!